packages/compass/src/app/components/home.tsx

```diff
@@ -281,122 +253,37 @@ function Home({
-      function close() {
-        setIsWelcomeOpen(false);
-        if (showSettings) {
-          appRegistry.emit('open-compass-settings');
-        }
+    (showSettingsModal?: boolean) => {
+      setIsWelcomeOpen(false);
+      if (showSettingsModal) {
+        showSettings();
       }
-      void close();
     },
-    [setIsWelcomeOpen, appRegistry]
+    [setIsWelcomeOpen, showSettings]
   );
 
-  useEffect(() => {
-    function onAutoupdateStarted() {
-      openToast('update-download', {
-        variant: 'progress',
-        title: 'Compass update is in progress',
-      });
-    }
-    function onAutoupdateFailed() {
-      openToast('update-download', {
-        variant: 'warning',
-        title: 'Failed to download Compass update',
-        description: 'Downloading a newer Compass version failed',
-      });
-    }
-    function onAutoupdateSucess() {
-      openToast('update-download', {
-        variant: 'note',
-        title: 'Restart to start newer Compass version',
-        description: (
-          <div className={restartPromptToastStyles}>
-            <div>
-              Continuing to use Compass without restarting may cause some of the
-              features to not work as intended.
-            </div>
-            <div>
-              <Button
-                variant={ButtonVariant.Primary}
-                onClick={() => {
-                  void hadronIpc.ipcRenderer?.call(
-                    'autoupdate:update-download-restart-confirmed'
-                  );
-                }}
-              >
-                Restart Compass
-              </Button>
-            </div>
-          </div>
-        ),
-        dismissible: true,
-        onClose: () => {
-          void hadronIpc.ipcRenderer?.call(
-            'autoupdate:update-download-restart-dismissed'
-          );
-        },
-      });
-    }
-    hadronIpc.ipcRenderer?.on(
-      'autoupdate:update-download-in-progress',
-      onAutoupdateStarted
-    );
-    hadronIpc.ipcRenderer?.on(
-      'autoupdate:update-download-failed',
-      onAutoupdateFailed
-    );
-    hadronIpc.ipcRenderer?.on(
-      'autoupdate:update-download-success',
-      onAutoupdateSucess
-    );
-    return () => {
-      hadronIpc.ipcRenderer?.removeListener(
-        'autoupdate:update-download-in-progress',
-        onAutoupdateStarted
-      );
-      hadronIpc.ipcRenderer?.removeListener(
-        'autoupdate:update-download-failed',
-        onAutoupdateFailed
-      );
-      hadronIpc.ipcRenderer?.removeListener(
-        'autoupdate:update-download-success',
-        onAutoupdateSucess
-      );
-    };
-  }, []);
-
   const connectionStorage =
     __TEST_CONNECTION_STORAGE === undefined
       ? ConnectionStorage
       : __TEST_CONNECTION_STORAGE;
   return (
-    <FileInputBackendProvider
-      createFileInputBackend={electronFileInputBackendRef.current}
-    >
+    <FileInputBackendProvider createFileInputBackend={createFileInputBackend}>
```

Comment:
What is the goal of having this inside home? This is so electron specific, why can't we set it up outside of home component? That's sort of the point of having Providers in the app